app/models/report.rb

```diff
@@ -3,15 +3,55 @@
 class Report < ApplicationRecord
   belongs_to :user
   has_many :comments, as: :commentable, dependent: :destroy
+  has_many :active_mentions, class_name: 'Mention', foreign_key: 'mentioning_id', inverse_of: 'mentioning', dependent: :destroy
+  has_many :passive_mentions, class_name: 'Mention', foreign_key: 'mentioned_id', inverse_of: 'mentioned', dependent: :destroy
+  has_many :mentioning_reports, through: :active_mentions, source: :mentioned
+  has_many :mentioned_reports, through: :passive_mentions, source: :mentioning
 
   validates :title, presence: true
   validates :content, presence: true
 
+  after_save :save_mentions
+
   def editable?(target_user)
     user == target_user
   end
 
   def created_on
     created_at.to_date
   end
+
+  private
+
+  def save_mentions
+    mentioning_report_ids_diff = fetch_mentioning_report_ids_diff
+    add_ids = mentioning_report_ids_diff[:added_ids]
+    remove_ids = mentioning_report_ids_diff[:removed_ids]
+
+    add_mentions(add_ids) if add_ids.present?
+    remove_mentions(remove_ids) if remove_ids.present?
```

Comment:
```suggestion
    add_mentions(add_ids)
    remove_mentions(remove_ids)
```

どちらもidsが空の時はメソッドを読んでも実害はないので、条件分岐は不要かなーと思います。
（`present?`を付ける前＝常に実行されていたときも動いてたわけですし）

Reply:
> どちらもidsが空の時はメソッドを読んでも実害はないので、条件分岐は不要かなーと思います。
> （present?を付ける前＝常に実行されていたときも動いてたわけですし）

おっしゃる通り、idsが空の時にメソッドを読んでも実害はないので、条件分岐は不要ですね！
こちら修正しました。
3511139658cb6c992e3af90